Add concept `random` and concept exercise `maze-maker` (#613)
exercism/elm: exercises/concept/maze-maker/.docs/hints.md

## 1. Start with the end

- Use the [`Random.constant`][random-constant] function

## 2. You can't catch flies with vinegar

- Use the [`Random.uniform`][random-uniform] function for generating the treasures
- Use the [`Random.map`][random-map] function for wrapping the treasure

## 3. Branching out

- Use the [`Random.int`][random-int] function for generating a value between 2 and 4
- You can access this value to create a `Branch` with the [`Random.andThen`][random-andThen] function
- Use the [`Random.list`][random-list] function to generate the branches
- Use the [`Random.map`][random-map] function for wrapping the branches

## 4. Amazing mazes

- Use the [`Random.weighted`][random-weighted] function for generating an uneven distribution of outcomes
- Use the [`Random.lazy`][random-lazy] function for using a generator recursively
- Read more about defining recursive data structures [here][bad-recursion]
- If you end up defining a nested `Generator (Generator Maze)` generator, you can use [`Random.andThen`][random-andThen] together with the `identity` function to flatten it

## 5. We have to go deeper

> - Use the [`Random.uniform`][random-uniform] function for generating the dead ends or rooms in on depth `0`

Reviewer: ```suggestion
- Use the [`Random.uniform`][random-uniform] function for generating the dead ends or rooms at depth `0`
```

This looks like a typo to me, but not sure exactly what the wording should be

> - If you end up defining a nested `Generator (Generator Maze)` generator, you can use the [`Random.andThen`][random-andThen] function to flatten it

Reviewer: Maybe make this the same as the other similar comment, and mention `identity`.
